R/model.R

```diff
@@ -506,6 +506,114 @@ compile_method <- function(quiet = TRUE,
 }
 CmdStanModel$set("public", name = "compile", value = compile_method)
 
+#' Check syntax of a Stan program
+#'
+#' @name model-method-syntax-check
+#' @aliases syntax_check
+#' @family CmdStanModel methods
+#'
+#' @description The `$syntax_check()` method of a [`CmdStanModel`] object
+#'   checks the Stan syntax of the model and returns `TRUE` if the compilation
+#'   succeeds. In case of a syntax error, an error is thrown.
+#'   
+#' @section Usage:
+#'   ```
+#'   $syntax_check(
+#'     quiet = TRUE,
+#'     include_paths = NULL,
+#'     stanc_options = list()
+#'   )
+#'   ```
+#'
+#' @section Arguments:
+#'   * `quiet`: (logical) Should the compile command be printed.
+#'   * `include_paths`: (character vector) Paths to directories where Stan
+#'   should look for files specified in `#include` directives in the Stan
+#'   program.
+#'   * `stanc_options`: (list) Any Stan-to-C++ transpiler options to be used
+#'   when compiling the model. See the **Examples** section below as well as the
+#'   `stanc` chapter of the CmdStan Guide for more details on available options:
+#'   https://mc-stan.org/docs/cmdstan-guide/stanc.html.
```

Comment:
The **Examples** section referred to here is actually in the doc for the compile method. Maybe change to: 

```suggestion
#'   * `stanc_options`: (list) Any Stan-to-C++ transpiler options to be used
#'   when compiling the model. See the documentation for the
#'   [`$compile()`][model-method-compile] method for details.
```